### ISSUE TITLE
Log multiple image plots per step

### ISSUE THREAD

**Issue by shcheklein** [iterative/dvclive#425]
Related and unblocks https://github.com/iterative/vscode-dvc/issues/1640

## Motivation

Allowing to display "per-step" images and plots is a relevant scenario for DL and is supported by TB, and other tools. In our case we can achieve this via checkpoints now (there is no UX for this), but also it would be good to support this w/o checkpoints that are involved to setup.

## Some ideas:

@daavoo:

Correct, but it would still be easy to do by formatting the output name:

```python
# Current default
live.log_image(f"image.png", image)
# Manually create structure
live.log_image(f"image/{live.step}.png", image)
```

@dberenbaum:

One option is to add a dvc.yaml flag like group/collapse/agg:

```yaml
plots:
  - dvclive/plots/images/image_dir:
      group: true
```

Any ideas for how to handle these in dvc plots diff (across all the products)?

Edit: Not sure how dvclive would know when to configure this given the current way of saving step-based images though 🤔

## Next steps

- [ ] Come up with DVC and DVCLive syntax / directory structure
- [ ] Update DVCLive (and DVC if needed)
- [ ] Come up with an interface in VS Code https://github.com/iterative/vscode-dvc/issues/1640
- [ ] Come up with an interface in Studio

## Questions

- How is it related to live metrics, do we want to send images per step?

**Comment by dberenbaum** [comment]
Coming back to this.

> * Come up with DVC and DVCLive syntax / directory structure

Seems like we are pretty committed to a `image/{live.step}.png` structure since we are continuing to move forward with no-checkpoint live metrics.

> * Update DVCLive (and DVC if needed)

The proposed solution from @daavoo above only requires documentation changes in DVCLive. It would be a breaking change to modify the default directory structure, and we already flip-flopped on this before in #326, so let's go with the docs changes @daavoo suggested.

On the DVC side, I suggested:

> @dberenbaum:
> 
> One option is to add a dvc.yaml flag like group/collapse/agg:
> 
> ```yaml
> plots:
>   - dvclive/plots/images/image_dir:
>       group: true
> ```

TBH I don't really like this 😄. The benefit of configuring in DVC is that users don't have to reconfigure across both VS Code and Studio or worry about losing these settings, but if it's only going to be a binary option in DVC, it doesn't seem like it saves the user much and feels inflexible compared to choosing how to view it in the GUI.

> Come up with an interface in VS Code

> * Come up with an interface in Studio

Can we agree on the previous questions, and then we can discuss the interfaces for those?

**Comment by daavoo** [comment]
> TBH I don't really like this 😄

Agree. I think it should be transparent for the user and the UIs should detect if a dir matches the structure described above and render the contents of the dir as a "slider", by default. 

> Can we agree on the previous questions, and then we can discuss the interfaces for those?

Unfortunately, I don't think the slider can be implemented with vega-lite so it would need to be implemented in the VSCode / Studio Frontend.

**Comment by shcheklein** [comment]
How about this idea.

- generate a JSON file with each image or one for the whole group
- DVC tools can recognize it and get additional info about the images
- besides this specific need (step number for an image), it can include other things that we can show as a tooltip (e.g. in case of the hackathon repo it would be nice to show inference values, what is the actual class, etc)

This way we don't have to configure the DVC pipeline + we can give a bit more general mechanism to enrich the information about the image plots.

**Comment by shcheklein** [comment]
> Do you have concerns starting with the plan of documenting image/{live.step}.png and coming up with an interface in VS Code and Studio to show a slider for them?

no major concerns with that, only if we decide to introduce changes to the dvc.yaml - that's where I would gravitate towards something else. It was an alternative to that.

If we are not introducing this, then it depends on the convention itself and how easy, hard it would be to implement w/o providing some additional signals.

**Comment by dberenbaum** [comment]
Discussed with @shcheklein and @mattseddon. Are we all aligned with these next steps?

* Document the pattern of `image/{live.step}.png` in dvclive (we can do this now)
* Start trying out in VS Code or Studio (is there a preference here?) how to collapse images for a directory structured this way and show a slider (probably a priority for next quarter unless it blocks example-get-started-experiments)

**Comment by shcheklein** [comment]
Yes, sounds good to me @dberenbaum !

**Comment by dberenbaum** [comment]
> should the convention be `image/{live.step}.png` or `{live.step}/image.png`

Let's move forward with `image/{live.step}.png` to unblock this

**Comment by daavoo** [comment]
> > should the convention be `image/{live.step}.png` or `{live.step}/image.png`
> 
> Let's move forward with `image/{live.step}.png` to unblock this

There is no real action to be made in DVCLive beyond documenting with an example. 
To move forwards we need to add support for recognizing the pattern and having a component to render (i.e. slider) in the dvc-render / VSCode / Studio.

**Comment by dberenbaum** [comment]
Yes, sorry @daavoo , @julieg18 is starting work in VS Code, which is why it came up. Let's keep this open as a docs reminder (or move it to dvc.org).

**Comment by dberenbaum** [comment]
Bumping back to p1 now that https://github.com/iterative/vscode-dvc/pull/4372 looks nearly ready to merge

**Comment by dberenbaum** [comment]
Closing in favor of https://github.com/iterative/dvc.org/issues/4746 (don't think there is any scope within dvclive itself right now).
